mixer/adapter/stackdriver/log/log.go

```diff
@@ -226,20 +227,35 @@ func toLabelMap(names []string, variables map[string]interface{}, logEntryTypes
 		v := variables[name]
 		switch vt := v.(type) {
 		case string:
-			out[name] = vt
+			out[name] = stripBadUTF8(vt)
 		case []byte:
 			if logEntryTypes[name] == istio_policy_v1beta1.IP_ADDRESS {
 				out[name] = net.IP(vt).String()
 			} else {
-				out[name] = fmt.Sprintf("%v", vt)
+				out[name] = stripBadUTF8(fmt.Sprintf("%v", vt))
 			}
 		default:
-			out[name] = fmt.Sprintf("%v", vt)
+			out[name] = stripBadUTF8(fmt.Sprintf("%v", vt))
 		}
 	}
 	return out
 }
 
+func stripBadUTF8(input string) string {
+	if utf8.ValidString(input) {
+		return input
+	}
+
+	out := make([]rune, 0, len(input))
+	for _, r := range input {
+		if r == utf8.RuneError {
+			continue
```

Comment:
Should there be an indication of this xform in the string?
Perhaps replace it with a character?

Reply:
I think it is better to just strip it? Implicitly transforming to a char could lead to confusion comparing to just removing it.